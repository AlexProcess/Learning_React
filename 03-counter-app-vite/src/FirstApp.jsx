import React from "react";
import { PropTypes } from "prop-types";

export const FirstApp = (props) => {
    const { title = "Hola soy Alex" } = props;
    const { subTitle = "Soy un subtitulos" } = props;
    const { nombre = "" } = props;

    // if (!title) {
    //     throw new Error("El titulo no existe");
    // }

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{nombre}</p>
        </>
    );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    nombre: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
    title: "No hay un titulo",
    subTitle: "No hay un subtitulo",
    nombre: "Alex Tercero",
};
