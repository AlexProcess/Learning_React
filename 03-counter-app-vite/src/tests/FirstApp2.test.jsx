import { getAllByText, render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe("Pruebas en FirstApp.jsx", () => {
    const title = "Hola, soy goku";
    const subTitle = "Soy un subtitulo";

    test("La PRIMERA PRUEBA DEBE DE HACER MATCH CON SNAPSHOT", () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });
    test("debe de mostrar el mensaje'Hola, soy Gokuy'", () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test("debe de mostrar un titulo que este en un h1", () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
            title
        );
    });

    test("debe de mostrar el subtitulo enviado por props", () => {
        render(<FirstApp title={title} subTitle={subTitle} />);

        expect(screen.getAllByText(subTitle).length).toBe(2);
    });
});
