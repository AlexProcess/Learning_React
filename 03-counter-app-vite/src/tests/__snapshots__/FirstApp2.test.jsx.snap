// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp.jsx La PRIMERA PRUEBA DEBE DE HACER MATCH CON SNAPSHOT 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy goku
  </h1>
  <p>
    No hay un subtitulo
  </p>
  <p>
    No hay un subtitulo
  </p>
  <p>
    Alex Tercero
  </p>
</div>
`;
